qt/scientific_interfaces/ISISReflectometry/GUI/Instrument/InstrumentPresenter.cpp

```diff
@@ -42,6 +44,20 @@ void InstrumentPresenter::notifyRestoreDefaultsRequested() {
   restoreDefaults();
 }
 
+void InstrumentPresenter::notifyEditingCalibFilePathFinished() {
+  auto const calibrationFilePath = m_view->getCalibrationFilePath();
+  if (!calibrationFilePath.empty()) {
+    try {
+      auto pocoPath = Poco::Path().parseDirectory(calibrationFilePath);
```

Comment:
Similarly there's an `IFileHandler` interface - I don't think it currently has functionality to check if a file exists but I think that would be a good place to put these checks; then if they're needed in other parts of the GUI we can re-use them via the same interface, and easily mock them etc.

To my shame, the concrete type for both of these interfaces is the `QtMainWindow`! They should really be separated out into separate classes. But because we're using the interfaces, that should be a pretty easy maintenance job to do at some point.

Reply:
Sounds good, thanks, I'll make that change. It looks like aren't unit tests for the FileHandler methods at the moment, I assume because they're defined in the view, or maybe because the nature of the existing methods would be hard to test? Assuming I haven't missed them somewhere, adding unit tests for the new method I'll add is something that I guess we could do as part of moving things into their own classes.